lib/Differentiator/ActivityAnalyzer.h

```diff
@@ -0,0 +1,77 @@
+#ifndef CLAD_DIFFERENTIATOR_ACTIVITYANALYZER_H
+#define CLAD_DIFFERENTIATOR_ACTIVITYANALYZER_H
+
+#include "clang/AST/RecursiveASTVisitor.h"
+#include "clang/Analysis/CFG.h"
+
+#include "clad/Differentiator/CladUtils.h"
+#include "clad/Differentiator/Compatibility.h"
+
+#include <algorithm>
+#include <iterator>
+#include <memory>
+#include <set>
+#include <utility>
+/// @brief Class that implemets Varied part of the Activity analysis.
+/// By performing static data-flow analysis, so called Varied variables
+/// are determined, meaning variables that depend on input parameters
+/// in a differentiable way. That result enables us to remove redundant
+/// statements in the reverse mode, improving generated codes efficiency.
+namespace clad {
+class VariedAnalyzer : public clang::RecursiveASTVisitor<VariedAnalyzer> {
+
+  bool m_Varied = false;
+  bool m_Marking = false;
+
+  std::set<const clang::VarDecl*>& m_VariedDecls;
+  using VarsData = std::set<const clang::VarDecl*>;
+  /// @brief A helper method to allocate VarsData
+  /// @param toAssign Parameter to initialize new VarsData with.
+  /// @return Unique pointer to a new object of type Varsdata.
```

Comment:
```suggestion
  /// A helper method to allocate VarsData
  /// \param toAssign - Parameter to initialize new VarsData with.
  /// \return Unique pointer to a new object of type Varsdata.
```

This is the style we use. Can you update the rest?